src/Pretzel.Logic/Templating/Context/Data.cs

```diff
@@ -0,0 +1,203 @@
+using System.Collections.Generic;
+using DotLiquid;
+using Pretzel.Logic.Extensions;
+using System.IO;
+using System.IO.Abstractions;
+using System.Collections;
+using System;
+using YamlDotNet.RepresentationModel;
+using CsvHelper;
+using System.Linq;
+using System.Globalization;
+using System.Text;
+
+namespace Pretzel.Logic.Templating.Context
+{
+    public class Data : Drop
+    {
+        private readonly IFileSystem fileSystem;
+        private readonly string dataDirectory;
+        private System.Lazy<object> cachedResult;
+
+        public Data(IFileSystem fileSystem, string dataDirectory)
+        {
+            this.fileSystem = fileSystem;
+            this.dataDirectory = dataDirectory;
+        }
+
+        public override object this[object method]
+        {
+            get
+            {
+                var res = base[method];
+                if (res != null)
+                {
+                    return res;
+                }
+
+                if (cachedResult == null)
+                {
+                    cachedResult = new Lazy<object>(() =>
+                    {
+                        if (!fileSystem.Directory.Exists(dataDirectory))
+                        {
+                            return null;
+                        }
+
+                        object result;
+
+                        if (TryParseYaml(dataDirectory, method.ToString(), out result))
```

Comment:
Is it possible to have data providers based on extension, with each provider in his own class?

Reply:
Do you mean by importing via MEF? Or just as an cleanup?

Reply:
Just as a cleanup, I don't think that DI is necessary in this case.

Reply:
Done.